packages/dashmate/src/docker/getServiceListFactory.js

```diff
@@ -12,52 +20,32 @@ function getServiceListFactory() {
    * @return {Object[]}
    */
   function getServiceList(config) {
-    const services = [
-      {
-        name: 'dashmate_helper',
-        title: 'Dashmate Helper',
-        image: DASHMATE_HELPER_DOCKER_IMAGE,
-      },
-      {
-        name: 'core',
-        title: 'Core',
-        image: config.get('core.docker.image'),
-      },
-    ];
+    const envs = generateEnvs(config);
+    const configProfiles = getConfigProfiles(config);
+
+    return Object
+      .entries(composeFile.services)
+      .map(([serviceName, { image: serviceImage, labels, profiles: serviceProfiles }]) => {
+        const title = labels?.['org.dashmate.service.title'];
 
-    if (config.get('core.masternode.enable')) {
-      services.push({
-        name: 'sentinel',
-        title: 'Sentinel',
-        image: config.get('core.sentinel.docker.image'),
-      });
-    }
+        if (!title) {
+          throw new Error(`Label for dashmate service ${serviceName} is not defined`);
+        }
 
-    if (config.get('platform.enable')) {
-      services.push({
-        name: 'drive_abci',
-        title: 'Drive ABCI',
-        image: config.get('platform.drive.abci.docker.image'),
-      }, {
-        name: 'drive_tenderdash',
-        title: 'Drive Tenderdash',
-        image: config.get('platform.drive.tenderdash.docker.image'),
-      }, {
-        name: 'dapi_api',
-        title: 'DAPI API',
-        image: config.get('platform.dapi.api.docker.image'),
-      }, {
-        name: 'dapi_tx_filter_stream',
-        title: 'DAPI Transactions Filter Stream',
-        image: config.get('platform.drive.abci.docker.image'),
-      }, {
-        name: 'dapi_envoy',
-        title: 'DAPI Envoy',
-        image: config.get('platform.dapi.envoy.docker.image'),
-      });
-    }
+        // Use hardcoded version for dashmate helper
+        // Or parse image variable and extract version from the env
+        const image = serviceName === 'dashmate_helper'
+          ? DASHMATE_HELPER_DOCKER_IMAGE : envs[serviceImage.match(new RegExp(/([A-Z])\w+/))[0]];
```

Comment:
Why do we use RegExp here? Just to trim whitespace? Why not `String#trim()`?

Reply:
Because when you read the compose file, it will read it as is, f.e. `${CORE_DOCKER_IMAGE:?err}`

Reply:
yeah, I just wondering how this regexp works. Won't be `([A-Z_]+)` more efficient and readable?